boot/commands.go

```diff
@@ -0,0 +1,104 @@
+package boot
+
+import (
+	"github.com/wakeful-deployment/operator/directory"
+	"github.com/wakeful-deployment/operator/global"
+	"github.com/wakeful-deployment/operator/node"
+	"time"
+)
+
+func DetectOrBootConsul(state *State) error {
+	// TODO
+	return nil
+}
+
+func LoadStateFromFile(path string) *State {
+	state, err := ReadStateFromConfigFile(path)
+
+	if err != nil {
+		global.Transition(global.NewState(global.ConfigFailed, err))
+		return nil
+	}
+
+	return state
+}
+
+func Boot(state *State) {
+	var err error
+
+	global.Transition(global.NewState(global.Booting, nil))
+
+	err = DetectOrBootConsul(state)
+
+	if err != nil {
+		global.Transition(global.NewState(global.ConsulFailed, err))
+		return
+	}
+
+	currentNodeState, err := node.CurrentState()
+
+	if err != nil {
+		global.Transition(global.NewState(global.FetchingNodeStateFailed, err))
+		return
+	}
+
+	err = Normalize(state, currentNodeState)
+
+	if err != nil {
+		global.Transition(global.NewState(global.NormalizingFailed, err))
+		return
+	}
+
+	global.Transition(global.NewState(global.Booted, nil))
+}
+
+func Run(currentState *State, wait string) {
+	if global.CurrentState.NotEqual(global.Running) && global.CurrentState.NotEqual(global.Booted) {
```

Comment:
`global.CurrentState` and `currentState` are confusing. Let's rename one of them.